src/test/java/org/zalando/intellij/swagger/file/FileDetectorTest.java

```diff
@@ -72,4 +75,47 @@ public void thatIsSwaggerYamlFileBasedOnFileContent() throws Exception {
         assertTrue(fileDetector.isMainSwaggerYamlFile(yamlFile));
     }
 
+    @Test
+    public void thatYamlFileExtensionIsSwaggerContentCompatible() throws Exception {
+        VirtualFile yamlFile = mock(VirtualFile.class);
+
+        when(yamlFile.getName()).thenReturn("file.yaml");
+
+        assertTrue(fileDetector.isSwaggerContentCompatible(yamlFile));
+    }
+
+    @Test
+    public void thatYmlFileExtensionIsSwaggerContentCompatible() throws Exception {
+        VirtualFile yamlFile = mock(VirtualFile.class);
+
+        when(yamlFile.getName()).thenReturn("file.yml");
+
+        assertTrue(fileDetector.isSwaggerContentCompatible(yamlFile));
+    }
+
+    @Test
+    public void thatJsonFileExtensionIsSwaggerContentCompatible() throws Exception {
+        VirtualFile yamlFile = mock(VirtualFile.class);
+
+        when(yamlFile.getName()).thenReturn("file.json");
+
+        assertTrue(fileDetector.isSwaggerContentCompatible(yamlFile));
+    }
+
+    @Test
+    public void thatSimilarileExtensionAreNotSwaggerContentCompatible() throws Exception {
```

Comment:
`thatSimilarileExtensionAreNotSwaggerContentCompatible` to `thatSimilarFileExtensionsAreNotSwaggerContentCompatible`